addons/account_edi/models/account_move.py

```diff
@@ -226,35 +226,78 @@ def _serialize_python_dictionary(vals):
+                    'src_line_id': src_line,
+                    'tax_id': tax,
+                    'src_tax_id': self.env['account.tax'].browse(row['group_tax_id']) if row['group_tax_id'] else tax,
+                    'tax_repartition_line_id': tax_line.tax_repartition_line_id,
+                    'base_amount': row['base_amount'],
+                    'tax_amount': row['tax_amount'],
+                    'base_amount_currency': row['base_amount_currency'],
+                    'tax_amount_currency': row['tax_amount_currency'],
+                })
+            invoice_lines_tax_values_dict.default_factory = None
+            return invoice_lines_tax_values_dict
+
+        def compute_invoice_lines_tax_values_dict_from_compute_all(invoice_lines):
+            invoice_lines_tax_values_dict = {}
+            sign = -1 if self.is_inbound() else 1
+            for invoice_line in invoice_lines:
+                taxes_res = invoice_line.tax_ids.compute_all(
+                    invoice_line.price_unit * (1 - (invoice_line.discount / 100.0)),
+                    currency=invoice_line.currency_id,
+                    quantity=invoice_line.quantity,
+                    product=invoice_line.product_id,
+                    partner=invoice_line.partner_id,
+                    is_refund=invoice_line.move_id.move_type in ('in_refund', 'out_refund'),
+                )
+                tax_values_list = invoice_lines_tax_values_dict[invoice_line] = []
+                rate = abs(invoice_line.balance) / abs(invoice_line.amount_currency) if invoice_line.amount_currency else 0.0
+                for tax_res in taxes_res['taxes']:
+                    tax_values_list.append({
+                        'base_line_id': invoice_line,
+                        'tax_id': self.env['account.tax'].browse(tax_res['id']),
+                        'tax_repartition_line_id': self.env['account.tax.repartition.line'].browse(tax_res['tax_repartition_line_id']),
+                        'base_amount': sign * invoice_line.company_currency_id.round(tax_res['base'] * rate),
+                        'tax_amount': sign * invoice_line.company_currency_id.round(tax_res['amount'] * rate),
+                        'base_amount_currency': sign * tax_res['base'],
+                        'tax_amount_currency': sign * tax_res['amount'],
+                    })
+            return invoice_lines_tax_values_dict
+
+        # Decide whether we will use AccountTax.compute_all or AccountMoveLine._get_query_tax_details to compute
+        # the tax amounts for each invoice.
+        # By default: if using 'Round Locally', we use compute_all (which will give rounded amounts for each line)
+        #             if using 'Round Globally', we use _get_query_tax_details (which will allocate the total tax amount proportionately among the lines)
+        # This can be changed using the 'use_sql_query_tax_details' context key.
+        company = self[0].company_id if self else self.env.company_id
```

Comment:
You make a `self.ensure_one()` [in line 221](https://github.com/odoo/odoo/pull/84312/files#diff-50d4c0850ad46ccdda325a6de23e3232b5a1e42cfb04238b08188c0bf50132b4R221). So the `self[0].company_id if self` is useless no ?
```suggestion
        company = self.company_id or self.env.company_id
```

Reply:
Heh, yes that's right. And thank you as well for spotting the case where `self.company_id` is not set.